providers/implementations/encode_decode/encode_key2any.c

```diff
@@ -1430,6 +1478,29 @@ MAKE_ENCODER(x448, ecx, EVP_PKEY_ED448, SubjectPublicKeyInfo, pem);
 # endif
 #endif
 
+#ifndef OPENSSL_NO_ML_KEM
+MAKE_ENCODER(ml_kem_512, ml_kem, EVP_PKEY_ML_KEM_512, EncryptedPrivateKeyInfo, der);
```

Comment:
Dunno if you noticed, but in `MAKE_ENCODER`, `evp_type` isn't used at all...  its comment mentions a future refactoring, but it appears that this has already happened.

Reply:
I hadn't noticed, good to know for the future.
I did not elect to create NIDs specifically for use here, they're convenient elsewhere in the code...

Reply:
Following this insight, I put together a refactoring PR, #26360